packages/core/src/indexing/sqlite/chronological-order.ts

```diff
@@ -0,0 +1,184 @@
+import type { Knex } from 'knex'
+import type { DataSource } from 'typeorm'
+import * as uint8arrays from 'uint8arrays'
+import { StreamID } from '@ceramicnetwork/streamid'
+import type { BaseQuery, Page, Pagination } from '../types.js'
+import {
+  BackwardPaginationQuery,
+  ForwardPaginationQuery,
+  PaginationKind,
+  parsePagination,
+} from '../parse-pagination.js'
+import { UnsupportedOrderingError } from '../ordering.js'
+import { asTableName } from '../as-table-name.util.js'
+
+type Selected = { stream_id: string; last_anchored_at: number; created_at: number }
+
+abstract class Cursor {
+  /**
+   * Decode cursor from base64url as JSON.
+   */
+  static parse(cursor: string): any {
+    return JSON.parse(uint8arrays.toString(uint8arrays.fromString(cursor, 'base64url')))
+  }
+
+  /**
+   * base64url-encode cursor from +input+ object.
+   * Return `undefined` if +input+ object is `undefined` or `null`.
+   */
+  static stringify(input: any): string | undefined {
+    if (input) {
+      return uint8arrays.toString(uint8arrays.fromString(JSON.stringify(input)), 'base64url')
+    } else {
+      return undefined
+    }
+  }
+}
+
+/**
+ * Prepare chronological cursor.
+ */
+function asChronologicalCursor(
+  input: { created_at: number; last_anchored_at: number } | undefined
+) {
+  if (!input) return undefined
+  return { created_at: input.created_at, last_anchored_at: input.last_anchored_at }
+}
+
+const CHRONOLOGICAL_ORDER = [
+  { column: 'last_anchored_at', order: 'DESC', nulls: 'last' }, // (`last_anchored_at` is null) DESC
```

Comment:
Changed to `// (last_anchored_at is null) DESC, sorts nulls first`

Reply:
woah yeah that was super not clear to me, and I didn't really understand what that comment was trying to convey.

Why does setting `nulls: 'last'` mean the nulls come first?  That seems so backwards.  Is it because the order is `DESC` and that inverts the meaning of `nuls: 'last'` for some reason?

Reply:
Hope, `sorts nulls first` makes it clearer.

This `nulls` thing is quite peculiar. Vanilla `{column: 'foo', order: 'DESC'}` translates to an obvious `ORDER BY foo DESC`. If you add `nulls: first`, the corresponding SQL is `ORDER BY (foo is not null) DESC`, and `nulls: last` turn it into `ORDER BY (foo is null) DESC`. A database interprets `foo is null` or `foo is not null` as a boolean condition. So, how a database sorts booleans?

SQLite stores boolean as an integer. False = 0, True = 1. Sorting booleans DESC results in [True, False] order.

This means rows with `(last_anchored_at is null) DESC` are sorted in [last_anchored_at is null, last_anchored_at is not null] order, which means "nulls first".

Just in case, this behaviour is specific to SQLite, because of the way it treats booleans. Postgres, and I suspect most other RDBMs, behave properly.